test/multiplexed-session.ts

```diff
@@ -0,0 +1,326 @@
+      });
+    multiplexedSession = new MultiplexedSession(DATABASE);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('instantiation', () => {
+    it('should localize the database instance', () => {
+      assert.strictEqual(multiplexedSession.database, DATABASE);
+    });
+
+    it('should apply defaults', () => {
+      assert.strictEqual(multiplexedSession.refreshRate, 7);
+    });
+
+    it('should create a session object', () => {
+      assert.deepStrictEqual(multiplexedSession._multiplexedSession, null);
+    });
+
+    it('should inherit from EventEmitter', () => {
+      assert(multiplexedSession instanceof events.EventEmitter);
+    });
+  });
+
+  describe('createSession', () => {
+    let _createSessionStub;
+    let _maintainStub;
+
+    beforeEach(() => {
+      _maintainStub = sandbox.stub(multiplexedSession, '_maintain');
+      _createSessionStub = sandbox
+        .stub(multiplexedSession, '_createSession')
+        .resolves();
+    });
+
+    it('should create mux session', () => {
+      multiplexedSession.createSession();
+      assert.strictEqual(_createSessionStub.callCount, 1);
+    });
+
+    it('should start housekeeping', done => {
+      multiplexedSession.createSession();
+      setImmediate(() => {
+        try {
+          assert.strictEqual(_maintainStub.callCount, 1);
+          done();
+        } catch (err) {
+          done(err);
+        }
+      });
+    });
+
+    it('should not propagate instance and database not found errors for Multiplexed Session', () => {
+      const multiplexedSession = new MultiplexedSession(DATABASE);
+      const error = {
+        code: Status.NOT_FOUND,
+        message: 'Database not found',
+      } as grpc.ServiceError;
+      sandbox.stub(multiplexedSession, '_createSession').rejects(error);
+      try {
+        multiplexedSession.createSession();
+      } catch (e) {
+        shouldNotBeCalled();
+      }
+    });
+
+    it('should not propagate errors for Multiplexed Session when default credentials not set', () => {
+      const multiplexedSession = new MultiplexedSession(DATABASE);
+      const error = {
+        message: 'Could not load the default credentials',
+      } as grpc.ServiceError;
+      sandbox.stub(multiplexedSession, '_createSession').rejects(error);
+      try {
+        multiplexedSession.createSession();
+      } catch (e) {
+        shouldNotBeCalled();
+      }
+    });
+
+    it('should not propagate errors for Multiplexed Session when projectId not set', () => {
+      const multiplexedSession = new MultiplexedSession(DATABASE);
+      const error = {
+        code: Status.NOT_FOUND,
+        message: 'Unable to detect a Project Id in the current environment',
+      } as grpc.ServiceError;
+      sandbox.stub(multiplexedSession, '_createSession').rejects(error);
+      try {
+        multiplexedSession.createSession();
+      } catch (e) {
+        shouldNotBeCalled();
+      }
+    });
```

Comment:
These tests have code duplication. You can have a single test with a loop for all message. For eg: https://github.com/googleapis/nodejs-spanner/blob/main/test/spanner.ts#L2769